### DISCUSSION TITLE
Can I use the /asr endpoint directly? All I get is a null response back.

### DISCUSSION THREAD

**Discussion by jacobstein123** [McCloudS/subgen#100] [Q&A] [answered]
I'm trying to directly use the /asr endpoint like I could with the [whisper-asr-webservice](https://github.com/ahmetoner/whisper-asr-webservice), but all I get back is a response of `null` a couple seconds after sending the POST. The same request works fine when I change the port to the regular whisper-asr-webservice.

**Comment by McCloudS** [comment]
What’s your POST request?  I have some of the parameters ignored because bazaar doesn’t use them.

**Reply by jacobstein123** [reply]
Just a simple POST to /asr: 

```
curl --location 'http://<IP>:9002/asr?language=en' \
--form 'audio_file=@"/some/path/to/downloaded_audio.mp3"'
```

**Reply by McCloudS** [reply]
The problem is I expected (via Bazarr) to always be given a correctly sampled file at a 16000hz sample rate, so it fails.  I'll see if I can fix it later.  In the meantime, you would have to convert your file and pipe it into the call.  
`ffmpeg -i jfk.flac -ar 16000 -ac 1 -f wav - | curl --location 'http://192.168.1.111:9000/asr?language=en' \
--form 'audio_file=@-;type=audio/wav'`

**Reply by jacobstein123** [reply]
Ah I see, that makes sense. I'll play around with converting the audio file before hitting the endpoint.

**Comment by McCloudS** [comment]
Also, the /batch endpoint could work for your use case depending on the options you need to set.

**Reply by jacobstein123** [reply]
I'll give that a try, thank you!

**Comment by McCloudS** [comment, marked as answer]
I did some light testing and have /asr endpoint now behaving correctly with encode (meaning, no pre-preparation of the file).  I only tested a couple files and didn't have any issues.  If you're still around, give it a shot.

**Reply by jacobstein123** [reply]
Sorry for the delay, but this works now!
